File: Python/7py 2/sum_of_a_beach.py
"""
|--------------------------------------------------------------------------
| Sum of a Beach
|--------------------------------------------------------------------------
|
"""

def sum_of_a_beach(beach):
   res, x = 0, ["sand", "water", "fish", "sun"]
   for i in x: res += len(beach.lower().split(i))-1
   return res


from re import findall, IGNORECASE
def sum_of_a_beach(beach):
   return len(findall('Sand|Water|Fish|Sun', beach, IGNORECASE))


def sum_of_a_beach(beach):
   return sum(beach.lower().count(i) for i in ["sand", "water", "fish", "sun"])


print(sum_of_a_beach("SanD"), 1)
print(sum_of_a_beach("sunshine"), 1)
print(sum_of_a_beach("sunsunsunsun"), 4)
print(sum_of_a_beach("123FISH321"), 1)

File: PHP/7php 2/sumOfABeach.php
<?php
/*
|--------------------------------------------------------------------------
| Sum of a Beach
|--------------------------------------------------------------------------
|
*/

function sumOfABeach($beach) {
   $res = 0; $x = ["sand", "water", "fish", "sun"];
   foreach ($x as $i) {
      $res += count(explode($i, strtolower($beach)))-1;
   }
   return $res;
}


function sumOfABeach($beach) {
   return preg_match_all('/(sand|water|fish|sun)/i', $beach);
}


function sumOfABeach($beach) {
   return array_sum(array_map(function($x) use ($beach) {
      return substr_count(strtolower($beach), $x);
   }, ['sand','water','fish','sun']));
}


print_r(sumOfABeach("WAtErSlIde") . ' ' . 1 . '
');
print_r(sumOfABeach("GolDeNSanDyWateRyBeaChSuNN") . ' ' . 3 . '
');
print_r(sumOfABeach("gOfIshsunesunFiSh") . ' ' . 4 . '
');
print_r(sumOfABeach("cItYTowNcARShoW") . ' ' . 0 . '
');